rflx/converter/iana.py

```diff
@@ -0,0 +1,279 @@
+        self.type_size = type_size
+        self.always_valid = always_valid and len(self.enum_literals) < 2**self.type_size
+
+    def __str__(self) -> str:
+        formatted_enum_literals = ",\n".join(str(r) for r in self.enum_literals)
+        return (
+            f"{'':<3}type {self.type_name} is\n"
+            f"{'':<6}({formatted_enum_literals.lstrip()})\n"
+            f"{'':<3}with Size => {self.type_size}"
+            f"{', Always_Valid;' if self.always_valid else ';'}"
+            f"\n\n"
+        )
+
+
+class EnumLiteral:
+    def __init__(
+        self,
+        rflx_name: str,
+        rflx_value: str,
+        bit_length: int,
+        comments: Optional[List[Element]] = None,
+    ):
+        self.name = rflx_name
+        self.value = rflx_value
+        self.bit_length = bit_length
+        self.comment_list = comments or []
+        self.alternative_names: List[str] = [f"alternative_name = {self.name}"]
+
+    def join(self, duplicate: EnumLiteral, registry_name: str) -> None:
+        self.name = f"{registry_name}_{self.value.replace('#', '_')}"
+        self.comment_list.extend(duplicate.comment_list)
+        self.alternative_names.append(f"alternative_name = {duplicate.name}")
+
+    @property
+    def comment(self) -> List[str]:
+        comments = [
+            f"{c.tag[c.tag.index('}') + 1:]} = {c.text}"
+            if c.tag is not None and c.text is not None
+            else f"Ref: {c.attrib['data']}"
+            for c in self.comment_list
+        ]
+        if len(self.alternative_names) > 1:
+            comments.extend(self.alternative_names)
+
+        formatted_comments = [
+            textwrap.wrap(comment, width=80, subsequent_indent=f"{'':<7}--  ")
+            for comment in comments
+        ]
+        if len(formatted_comments) > 0:
+            return ["\n".join(formatted_comment) for formatted_comment in formatted_comments]
+        return []
+
+    def __str__(self) -> str:
+        str_repr = ""
+        comment = self.comment
+        if comment:
+            str_repr += "\n"
+            for comment_line in comment:
+                str_repr += f"{'':<7}-- {comment_line}\n"
+        name = (
+            self.name
+            if len(self.name) <= 100
+            else self.name[:100]
+            + ("_" if self.name[99] != "_" else "")
+            + self.value.replace("#", "_")
+        )
+        str_repr += f"{'':<7}{f'{name} => {self.value}'}"
+        return str_repr
+
+
+def _normalize_name(description_text: str) -> str:
+    t: Dict[str, Union[int, str, None]] = {c: " " for c in string.punctuation + "\n"}
+    name = description_text.translate(str.maketrans(t))
+    name = "_".join([s[0].upper() + s[1:] for s in name.split()])
+    return name
+
+
+def _normalize_value(value: str) -> Tuple[str, int]:
+    # in case range is defined, return the upper bound of that range
+    # ranges can only occur for UNASSIGNED or RESERVED values wich are filtered out
```

Comment:
How about adding a docstring here instead of the comment?

Reply:
Done.